packages/decision-tree/src/DecitionTree.test.tsx

```diff
@@ -0,0 +1,76 @@
+import React from 'react';
+import { DecisionTree } from './DecisionTree';
+import { renderWithProviders, screen, fireEvent, waitFor } from '@tests/renderWithProviders';
+import { questionnaire, serverMockup } from './tests/Questionnaire.mock';
+import { UseSetupCallbackCB } from '@types';
+
+describe('DecisionTree', () => {
+  const callback = async function (questionId, value) {
+    console.log(questionId, value);
+    return serverMockup[questionId];
+  } as UseSetupCallbackCB;
+
+  test('should render all the components', () => {
+    renderWithProviders(<DecisionTree questionnaire={questionnaire} callback={callback} />);
+    expect(screen.getAllByText('Next')).toHaveLength(1);
+    expect(screen.getAllByText('Back')).toHaveLength(1);
+    expect(screen.getAllByText('Male')).toHaveLength(1);
+    expect(screen.getAllByText('Female')).toHaveLength(1);
+    expect(screen.getAllByText('What sex were you assigned at birth?')).toHaveLength(1);
+  });
+
+  test('should disable both buttons by default', () => {
+    renderWithProviders(<DecisionTree questionnaire={questionnaire} callback={callback} />);
+    expect(screen.getByText('Back').parentElement).toBeDisabled();
+    expect(screen.getByText('Next').parentElement).toBeDisabled();
+  });
+
+  test('should enable next button on value select', () => {
+    renderWithProviders(<DecisionTree questionnaire={questionnaire} callback={callback} />);
+
+    fireEvent.click(screen.getByText('Female'));
+
+    expect(screen.getByText('Next').parentElement).not.toBeDisabled();
```

Comment:
why are we referencing the parent element?

Reply:
because is the element being disabled. I think it is because the button structure, like there is a span with the text and the button on top.

Reply:
can we use `getByRole('button', { name: 'Next' })` instead? In general this is the way we are referencing components in unit tests, and if I remember properly it is the way react testing library recommends to do it

Reply:
ok, this works, but I cannot figure out why it works, I'm selecting through role, html attribute which is note set as we removed it, and then it uses the name property to select text?? could you tell me why it works so I understand please??

Reply:
this is the recommended way, it uses accessibility properties to reference the component. You can read more in here:

- https://kentcdodds.com/blog/common-mistakes-with-react-testing-library#using-the-wrong-query
- https://testing-library.com/docs/queries/about/#priority

Reply:
TIL, it is using the accessibility tree which converts text to name properties.